packages/vuetify/src/components/VSelect/VSelect.tsx

```diff
@@ -0,0 +1,231 @@
+// Styles
+import './VSelect.sass'
+
+// Components
+import { VChip } from '@/components/VChip'
+import { VIcon } from '@/components/VIcon'
+import { VList, VListItem } from '@/components/VList'
+import { VMenu } from '@/components/VMenu'
+import { VTextField } from '@/components/VTextField'
+
+// Composables
+import { makeFilterProps, useFilter } from '@/composables/filter'
+import { useForwardRef } from '@/composables/forwardRef'
+import { useLocale } from '@/composables/locale'
+import { useProxiedModel } from '@/composables/proxiedModel'
+
+// Utility
+import { computed, ref, watch } from 'vue'
+import { genericComponent, useRender, wrapInArray } from '@/util'
+
+// Types
+import type { LinkProps } from '@/composables/router'
+import type { MakeSlots } from '@/util'
+import type { PropType } from 'vue'
+
+export type SelectItem = string | (string | number)[] | ((item: Record<string, any>, fallback?: any) => any) | (LinkProps & {
+  text: string
+})
+
+export const VSelect = genericComponent<new <T>() => {
+  $slots: MakeSlots<{
+    default: []
+    title: []
+  }>
+}>()({
+  name: 'VSelect',
+
+  props: {
+    chips: Boolean,
+    hideNoData: Boolean,
+    hideSelected: Boolean,
+    items: {
+      type: Array as PropType<SelectItem[]>,
+      default: () => ([]),
+    },
+    modelValue: {
+      type: [String, Array],
+      default: () => ([]),
+    },
+    multiple: Boolean,
+    noDataText: {
+      type: String,
+      default: '$vuetify.noDataText',
+    },
+    openOnClear: Boolean,
+
+    ...makeFilterProps(),
+  },
+
+  emits: {
+    'click:clear': (e: MouseEvent) => true,
+    'update:modelValue': (val: any) => true,
+  },
+
+  setup (props, { attrs, slots }) {
+    const { t } = useLocale()
+    const vTextFieldRef = ref()
+    const activator = ref()
+    const model = useProxiedModel(
+      props,
+      'modelValue',
+      [],
+      v => wrapInArray(v),
+      (v: any) => props.multiple ? v : v[0]
+    )
+    const { filteredItems } = useFilter(props, props.items)
```

Comment:
I still don't understand why we're filtering on nothing here

Reply:
So that we are working with the same data structure that other selects will have.